src/auth/auth-api-request.ts

```diff
@@ -36,6 +36,7 @@ import {
 } from './auth-config';
 import { Tenant, TenantServerResponse } from './tenant';
 import { auth } from './index';
+import { useEmulator } from './auth';
```

Comment:
This creates a cyclic import. Not a major problem, but something we should avoid if possible. I think one solution is to export this function from this module, and import it into the `auth` module. Another is to invoke the function in `auth`, and pass the resulting boolean value in to the classes in this module.

Also note that the constructor of `AuthResourceUrlBuilder` (in this module) also currently does the env variable check. Perhaps refactor that out with the same function while we are at it.

Reply:
+1 to avoid cyclic import and to @hiranya911's suggestions.

Reply:
Good idea, moved and unified.